pipelines/build/common/openjdk_build_pipeline.groovy

```diff
@@ -394,38 +396,57 @@ class Build {
 
     MetaData formMetadata(VersionInfo version, Boolean initialWrite) {
 
+        // We have to setup some attributes for the first run since formMetadata is sometimes initiated from downstream job on master node with no access to the required files
         if (initialWrite) {
-            context.println "FIRST METADATA WRITE OUT! Checking if we have a scm reference in the build config..."
+
+            // Get scmRef
+            context.println "INFO: FIRST METADATA WRITE OUT! Checking if we have a scm reference in the build config..."
 
             String scmRefPath = "workspace/target/scmref.txt"
             scmRef = buildConfig.SCM_REF
 
             if (scmRef != "") {
                 // Use the buildConfig scmref if it is set
-                context.println "SCM_REF has been set (${buildConfig.SCM_REF})! Using it to build the inital metadata over ${scmRefPath}..."
+                context.println "SUCCESS: SCM_REF has been set (${buildConfig.SCM_REF})! Using it to build the inital metadata over ${scmRefPath}..."
             } else {
                 // If we don't have a scmref set in config, check if we have a scmref from the build
-                context.println "SCM_REF is NOT set. Attempting to read ${scmRefPath}..."
+                context.println "INFO: SCM_REF is NOT set. Attempting to read ${scmRefPath}..."
                 try {
                     scmRef = context.readFile(scmRefPath).trim()
-                    context.println "scmref.txt found: ${scmRef}"
+                    context.println "SUCCESS: scmref.txt found: ${scmRef}"
                 } catch (NoSuchFileException e) {
                     // In rare cases, we will fail to create the scmref.txt file
                     context.println "WARNING: $scmRefPath was not found. Using build config SCM_REF instead (even if it's empty)..."
                 }
 
             }
 
+            // Get full version output
+            String versionPath = "workspace/target/version.txt"
+            context.println "INFO: Attempting to read ${versionPath}..."
+
+            // No try catch on this as we should always have a version.txt file
+            fullVersionOutput = context.readFile(versionPath)
```

Comment:
I think you'll want some defensive programming here, you never know......

Reply:
Done